core/src/main/kotlin/io/customer/sdk/lifecycle/CustomerIOActivityLifecycleCallbacks.kt

```diff
@@ -18,9 +19,13 @@ import kotlinx.coroutines.launch
  * and should avoid listening to events from ActivityLifecycleCallbacks directly.
  */
 class CustomerIOActivityLifecycleCallbacks : Application.ActivityLifecycleCallbacks {
-    private val lifecycleEvents = MutableSharedFlow<LifecycleStateChange>(replay = 1)
+    private val lifecycleEvents = MutableSharedFlow<LifecycleStateChange>()
     private val subscriberScope = SDKComponent.scopeProvider.lifecycleListenerScope
 
+    // Stores last emitted lifecycle state to provide current state to new subscribers.
+    // The reference is weak to avoid memory leaks.
+    private var currentLifecycleStateRef: WeakReference<LifecycleStateChange>? = null
```

Comment:
This is only added for wrappers. If we want to skip it for now and update later when working on wrappers, I'm okay with it.